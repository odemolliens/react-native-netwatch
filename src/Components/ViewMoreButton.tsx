import * as React from 'react';
import { useContext } from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { ThemeContext } from '../Theme';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

export interface IProps {
  onPress: () => void;
}

// Button use in the details screens to go in the JSON tree view. It's just a button text.
export const ViewMoreButton: React.FC<IProps> = (props: IProps) => {
  const theme = useContext(ThemeContext);

  return (
    <TouchableOpacity
      testID="buttonViewMore"
      style={styles.container}
      onPress={() => {
        props.onPress();
      }}
    >
      <Text style={[{ color: theme.primaryColor, marginRight: 6 }]}>View more</Text>
      <MaterialCommunityIcons
        name="more"
        color={theme.primaryColor}
        size={14}
        style={{ transform: [{ rotateY: '180deg' }] }}
      />
    </TouchableOpacity>
  );
};

export default ViewMoreButton;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderLeftWidth: 0,
    alignItems: 'center',
  },
});
