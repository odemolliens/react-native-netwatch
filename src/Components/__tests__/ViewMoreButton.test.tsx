import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ViewMoreButton, IProps } from '../ViewMoreButton';

describe('Status test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;
  const onPress = jest.fn();

  it('should render properly redux action', () => {
    givenProps(onPress);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<ViewMoreButton {...props} />);
  }

  function givenProps(onPress: () => void) {
    props = {
      onPress,
    };
  }
});
