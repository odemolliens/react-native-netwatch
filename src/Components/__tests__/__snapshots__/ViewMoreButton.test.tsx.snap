// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status test suite should render properly redux action 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderLeftWidth": 0,
      "flexDirection": "row",
    }
  }
  testID="buttonViewMore"
>
  <Component
    style={
      Array [
        Object {
          "color": "#06B6D4",
          "marginRight": 6,
        },
      ]
    }
  >
    View more
  </Component>
  <Icon
    allowFontScaling={false}
    color="#06B6D4"
    name="more"
    size={14}
    style={
      Object {
        "transform": Array [
          Object {
            "rotateY": "180deg",
          },
        ],
      }
    }
  />
</ForwardRef>
`;
